packages/llamaindex/src/embeddings/MixedbreadAIEmbeddings.ts

```diff
@@ -0,0 +1,140 @@
+import { BaseEmbedding } from "./types.js";
+
+type EmbeddingsRequestWithoutInput = Omit<
+  MixedbreadAI.EmbeddingsRequest,
+  "input"
+>;
+
+/**
+ * Interface extending EmbeddingsParams with additional
+ * parameters specific to the MixedbreadAIEmbeddings class.
+ */
+export interface MixedbreadAIEmbeddingsParams
+  extends Omit<EmbeddingsRequestWithoutInput, "model"> {
+  /**
+   * The model to use for generating embeddings.
+   * @default {"mixedbread-ai/mxbai-embed-large-v1"}
+   */
+  model?: string;
+
+  /**
+   * The maximum number of documents to embed in a single request.
+   * @default {128}
+   */
+  batchSize?: number;
+
+  /**
+   * The API key to use.
+   * @default {process.env.MXBAI_API_KEY}
+   */
+  apiKey?: string;
+
+  /**
+   * The base URL for the API.
+   */
+  baseUrl?: string;
+}
+
+/**
+ * Class for generating embeddings using the mixedbread ai API.
+ *
+ * This class leverages the model "mixedbread-ai/mxbai-embed-large-v1" to generate
+ * embeddings for text documents. The embeddings can be used for various NLP tasks
+ * such as similarity comparison, clustering, or as features in machine learning models.
+ *
+ * @example
+ * const mxbai = new MixedbreadAIEmbeddings({ apiKey: 'your-api-key' });
+ * const texts = ["Baking bread is fun", "I love baking"];
+ * const result = await mxbai.getTextEmbeddings(texts);
+ * console.log(result);
+ *
+ * @example
+ * const mxbai = new MixedbreadAIEmbeddings({
+ *  apiKey: 'your-api-key',
+ *  model: 'mixedbread-ai/mxbai-embed-large-v1',
+ *  encodingFormat: MixedbreadAI.EncodingFormat.Binary,
+ *  dimensions: 512,
+ *  normalized: true,
+ * });
+ * const query = "Represent this sentence for searching relevant passages: Is baking bread fun?";
+ * const result = await mxbai.getTextEmbedding(query);
+ * console.log(result);
+ */
+export class MixedbreadAIEmbeddings extends BaseEmbedding {
+  requestParams: EmbeddingsRequestWithoutInput;
+  private client: MixedbreadAIClient;
+
+  /**
+   * Constructor for MixedbreadAIEmbeddings.
+   * @param {Partial<MixedbreadAIEmbeddingsParams>} params - An optional object with properties to configure the instance.
+   * @throws {Error} If the API key is not provided or found in the environment variables.
+   * @throws {Error} If the batch size exceeds 256.
+   */
+  constructor(params?: Partial<MixedbreadAIEmbeddingsParams>) {
+    super();
+
+    const apiKey = params?.apiKey ?? getEnv("MXBAI_API_KEY");
+    if (!apiKey) {
+      throw new Error(
+        "mixedbread ai API key not found. Either provide it in the constructor or set the 'MXBAI_API_KEY' environment variable.",
+      );
+    }
+    if (params?.batchSize && params?.batchSize > 256) {
+      throw new Error(
+        "The maximum batch size for mixedbread ai embeddings API is 256.",
+      );
+    }
+
+    this.embedBatchSize = params?.batchSize ?? 128;
+    this.requestParams = {
+      model: params?.model ?? "mixedbread-ai/mxbai-embed-large-v1",
```

Comment:
use "default" or "latest" as well like in reranker?

Reply:
Doesn't make a lot of sense for embeddings from my perspective